main.go

```diff
@@ -19,22 +22,29 @@ var (
 )
 
 func main() {
+	err := mainWithError()
+	if err != nil {
+		panic(fmt.Sprintf("%#v\n", microerror.Mask(err)))
+	}
+}
+
+func mainWithError() error {
```

Comment:
We panic in main, a few lines above. It is just that we can properly inspect and print the error stack one time and not with every panic. And the error handling in this function can just be done as usual.